src/app/commons/MarkdownEditor.js

```diff
@@ -18,48 +17,32 @@ function Tabs({textareaVal, setTextareaVal}) {
             <div className="tabs is-boxed">
                 <ul>
                     <li className={toggleState === TAB_WRITE ? "is-active active-tabs" : ""}>
-                        <p className="tab-write"
-                           onClick={() => setToggleState(TAB_WRITE)}
-                        >
-                            Write
-                        </p>
+                        <p className="tab-write" onClick={() => setToggleState(TAB_WRITE)}>Write</p>
                     </li>
                     <li className={toggleState === TAB_PREVIEW ? "is-active active-tabs" : ""}>
-                        <p className="tab-preview"
-                           onClick={() => setToggleState(TAB_PREVIEW)}
-                        >
-                            Preview
-                        </p>
+                        <p className="tab-preview" onClick={() => setToggleState(TAB_PREVIEW)}>Preview</p>
                     </li>
                 </ul>
             </div>
-
             <div className="content-tabs">
                 <div className={(toggleState === TAB_WRITE ? "active-textarea" : "hide")}>
-                    <textarea
-                        className="description-textarea"
-                        value={textareaVal}
-                        onChange={handleTextareaChange}
-                        style={{backgroundColor: "white"}}
-                    />
+                    <textarea className="description-textarea"
+                              value={textareaVal}
+                              onChange={handleTextareaChange}
+                              style={{backgroundColor: "white"}}/>
                 </div>
-
-                <div className={(toggleState === TAB_PREVIEW ? "active-markdown" : "hide")}
-
-                >
-                    <ReactMarkdown>
-                        {textareaVal}
-                    </ReactMarkdown>
+                <div className={(toggleState === TAB_PREVIEW ? "active-markdown" : "hide")}>
+                    <ReactMarkdown>{textareaVal}</ReactMarkdown>
                 </div>
             </div>
         </div>
     );
 }
 
-const MarkdownEditor = ({style}) => {
+const MarkdownEditor = ({text, onSaved, style}) => {
     // TODO: problemService.getProblemDescription
     const [editingState, setEditingState] = useState(false);
-    const [textareaVal, setTextareaVal] = useState('Press Edit Description to start writing the description. Styling with Markdown is supported.\n');
+    const [textareaVal, setTextareaVal] = useState(text ? text : 'Press Edit Description to start writing the description. Styling with Markdown is supported.\n');
```

Comment:
`text || 'Press Edit Description to start writing the description. Styling with Markdown is supported.\n'`

Reply:
I gotta say this statement is truly interesting .

Reply:
Does this work?

Reply:
It does work.